src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -689,6 +708,163 @@ MarkNodeHasMetadata(char *nodeName, int32 nodePort, bool hasMetadata)
+ * commands are wrapped with a `worker_apply_sequence_command` call, which sets the
+ * sequence space uniquely for each worker.
+ */
+List *
+SequenceDDLCommandsForTable(Oid relationId)
+{
+	List *sequenceDDLList = NIL;
+	List *ownedSequences = getOwnedSequences(relationId);
+	ListCell *listCell;
+	char *ownerName = TableOwner(relationId);
+	bool hasMetadataWorkers = HasMetadataWorkers();
+
+	foreach(listCell, ownedSequences)
+	{
+		Oid sequenceOid = (Oid) lfirst_oid(listCell);
+		char *sequenceDef = pg_get_sequencedef_string(sequenceOid);
+		char *escapedSequenceDef = quote_literal_cstr(sequenceDef);
+		StringInfo wrappedSequenceDef = makeStringInfo();
+		StringInfo sequenceGrantStmt = makeStringInfo();
+		Oid schemaId = InvalidOid;
+		char *createSchemaCommand = NULL;
+		char *sequenceName = get_rel_name(sequenceOid);
+
+		if (hasMetadataWorkers)
+		{
+			EnsureSupportedSequenceColumnType(sequenceOid);
+
+			/* create schema if needed */
+			schemaId = get_rel_namespace(sequenceOid);
+			createSchemaCommand = CreateSchemaDDLCommand(schemaId);
+			if (createSchemaCommand != NULL)
+			{
+				sequenceDDLList = lappend(sequenceDDLList, createSchemaCommand);
+			}
+
+			appendStringInfo(wrappedSequenceDef,
+							 WORKER_APPLY_SEQUENCE_COMMAND,
+							 escapedSequenceDef);
+
+			appendStringInfo(sequenceGrantStmt,
+							 "GRANT ALL ON SEQUENCE %s TO %s", sequenceName, ownerName);
+
+			sequenceDDLList = lappend(sequenceDDLList, wrappedSequenceDef->data);
+			sequenceDDLList = lappend(sequenceDDLList, sequenceGrantStmt->data);
+		}
+	}
+
+	return sequenceDDLList;
+}
+
+
+/*
+ * CreateSchemaDDLCommand returns a "CREATE SCHEMA..." SQL string for creating the given
+ * schema if not exists and with proper authorization.
+ */
+char *
+CreateSchemaDDLCommand(Oid schemaId)
+{
+	char *schemaName = get_namespace_name(schemaId);
+	StringInfo schemaNameDef = NULL;
+	char *ownerName = NULL;
+
+	if (strncmp(schemaName, "public", NAMEDATALEN) == 0)
+	{
+		return NULL;
+	}
+
+	schemaNameDef = makeStringInfo();
+	ownerName = SchemaOwner(schemaId);
+	appendStringInfo(schemaNameDef, CREATE_SCHEMA_COMMAND, schemaName, ownerName);
+
+	return schemaNameDef->data;
+}
+
+
+/*
+ * EnsureSupportedSequenceColumnType looks at the column which depends on this sequence
+ * (which it Assert's exists) and makes sure its type is suitable for use in a disributed
+ * manner.
+ *
+ * Any column which depends on a sequence (and will therefore be replicated) but which is
+ * not a BIGINT cannot be used for an mx table, because there aren't enough values to
+ * ensure that generated numbers are globally unique.
+ */
+static void
+EnsureSupportedSequenceColumnType(Oid sequenceOid)
+{
+	Oid tableId = InvalidOid;
+	Oid columnType = InvalidOid;
+	int32 columnId = 0;
+	bool shouldSyncMetadata = false;
+	bool sequenceOwned = false;
+
+	sequenceOwned = sequenceIsOwned(sequenceOid, &tableId, &columnId);
+	if (!sequenceOwned)
+	{
+		aclcheck_error(ACLCHECK_NOT_OWNER, ACL_KIND_SEQUENCE,
```

Comment:
makes sense, added `pg_class_ownercheck`. We still need to keep `sequenceIsOwned` call though, in order to fetch the table and column id